### PR TITLE
Support mockgcp for BigQueryConnectionConnection

### PR DESCRIPTION
**Pull request by maqiuyujoyce**
### Change description

Support mockgcp to talk to BigQuery Connection v1 API.

### Tests you have done

- [N/A] Run `make ready-pr` to ensure this PR is ready for review.
- [X] Perform necessary E2E testing for changed resources.

`E2E_KUBE_TARGET=envtest RUN_E2E=1 E2E_GCP_TARGET=mock GOLDEN_REQUEST_CHECKS=1 GOLDEN_OBJECT_CHECKS=1 go test -test.count=1 -timeout 3600s -v ./tests/e2e -run TestAllInSeries/fixtures/bigqueryconnectionconnection` passed.

### CODE REVIEW

**Comment by justinsb** [comment]
Hmmm... why are the tests not running....

/ok-to-test

**Comment by justinsb** [comment]
Looks great.  I'm a little scared that the tests don't seem to be running though...

**Comment by maqiuyujoyce** [comment]
> Looks great. I'm a little scared that the tests don't seem to be running though...

The first run of the test suite never finish for some reason: https://github.com/GoogleCloudPlatform/k8s-config-connector/actions/runs/9999441455 I tried to cancel the workflow multiple times yesterday but it didn't work. I was wondering if it is the problem of GitHub...

**Comment by justinsb** [comment]
Yes, looks like github actions had a whoopsie yesterday: https://www.githubstatus.com/history

Want to try rebasing and force pushing?

**Comment by maqiuyujoyce** [comment]
Will do!

File: mockgcp/mockbigqueryconnection/connection.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	}
+
+	fqn := name.String()
+	now := time.Now()
+
+	obj := proto.Clone(req.Connection).(*pb.Connection)
+
+	// The returned "name" in the response uses the project number instead of
+	// project ID.
+	obj.Name = strings.ReplaceAll(fqn, "mock-project", "${projectNumber}")
```
[the remark this answers is not in the document]
Comment (maqiuyujoyce) [reply mockgcp/mockbigqueryconnection/connection.go]: Updated. Now it doesn't care what project ID it is.

**Comment by justinsb** [comment]
Awesome - thanks @maqiuyujoyce 

/approve
/lgtm

**Pull request merged**